api/controller.go

```diff
@@ -969,6 +969,10 @@ func (c *Controller) BranchesDiffBranchHandler() branches.DiffBranchHandler {
 		limit := int(swag.Int64Value(params.Amount))
 		after := swag.StringValue(params.After)
 		diff, hasMore, err := cataloger.DiffUncommitted(deps.ctx, params.Repository, params.Branch, limit, after)
+		if errors.Is(err, catalog.ErrNotFound) {
```

Comment:
We have TWO different ways not to find something: `catalog.ErrNotFound` and `graveler.ErrNotFound`.  Well, actually 3 because we also have one in `db`.  Elsewhere in this file we test for both alternatives; could you do so here, too?
One day we'll finish getting rid of package `catalog` and then we'll only have one way not to find something.

Reply:
Actually, it's safe to use `catalog.ErrNotFound` everywhere (`catalog.ErrNotFound` is actually `db.ErrNotFound`, while `graveler.ErrNotFound` is a wrapper around `db.ErrNotFound`, or in other words, `errors.Is(err, catalog.ErrNotFound)` will catch any of the 3). It's not pretty but it's temporary until we reduce the amount of layers.

Reply:
Cool.  But a yucky shade of cool, because it is clearly wrong.  Luckily `db` is going away soon, so we can keep this.